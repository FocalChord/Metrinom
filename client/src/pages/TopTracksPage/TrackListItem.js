import { Avatar, Box, Divider, ListItem, ListItemAvatar, ListItemText } from "@material-ui/core";
import React from "react";
import { useHistory } from "react-router-dom";

const TrackListItem = ({ track, idx }) => {
    const history = useHistory();

    return (
        <Box>
            <ListItem button onClick={() => history.push(`/track/${track.trackId}`)}>
                <ListItemAvatar>
                    <Avatar
                        style={{
                            borderRadius: 0,
                            width: 65,
                            height: 65,
                        }}
                        src={track.trackImage}
                    ></Avatar>
                </ListItemAvatar>
                <ListItemText
                    style={{
                        paddingLeft: 20,
                    }}
                    primary={track.trackName}
                    secondary={
                        <React.Fragment>
                            Artist - {track.artistName}
                            <br />
                            Album - {track.albumName}
                        </React.Fragment>
                    }
                />
                <div
                    style={{
                        fontSize: 25,
                    }}
                >
                    {idx + 1}
                </div>
            </ListItem>
            <Divider variant="inset" />
        </Box>
    );
};

export default TrackListItem;
