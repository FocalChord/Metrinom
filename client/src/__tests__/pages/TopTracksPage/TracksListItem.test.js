import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import TrackListItem from "../../../pages/TopTracksPage/TrackListItem";
import { MemoryRouter } from "react-router-dom";

test("<TrackListItem />", () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <TrackListItem track={{ name: "Godzilla" }} idx="1" />
            </MemoryRouter>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
