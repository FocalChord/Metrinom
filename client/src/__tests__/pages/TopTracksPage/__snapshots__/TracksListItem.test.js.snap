// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackListItem /> 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        style={
          Object {
            "borderRadius": 0,
            "height": 65,
            "width": 65,
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiListItemText-root"
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      <p
        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Artist - 
        <br />
        Album - 
      </p>
    </div>
    <div
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      11
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-inset"
  />
</div>
`;
